crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -118,6 +118,71 @@ impl Ellipse {
         (a * a - b * b).sqrt() / a
     }
 
+    #[inline(always)]
+    /// Get the focal length of the ellipse. This corresponds to the distance between one of the foci and the center of the ellipse.
+    pub fn focal_length(&self) -> f32 {
+        let a = self.semi_major();
+        let b = self.semi_minor();
+
+        (a * a - b * b).sqrt()
+    }
+
+    #[inline(always)]
```

Comment:
Do we want to always inline this function, considering it's a pretty long method?

Reply:
Most of it is just the `const` array and excluding that, the method really isn't that long. In the end, this is probably a design decision, so if you believe it should be removed, I am happy to do so